### PR TITLE
enh(upgrade): externalize release note documentation during upgrade

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

During the upgrade process we will now use the release notes that will be written in the new documentation.

The timeout on the button has been also dicreased from 10s to 3s

**Before**
from

**After**
to

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

* Upgrade to latest 20.04.x from 19.10.x for example
* New step 3 should be displayed

### CODE REVIEW

File: www/install/step_upgrade/step3.php [hunk 1]
```diff
@@ -43,55 +43,59 @@
 require_once '../steps/functions.php';
 $template = getTemplate('templates');
 
+require_once realpath(dirname(__FILE__) . "/../../../config/centreon.config.php");
```
Comment (kduret) [inline www/install/step_upgrade/step3.php:46]:
```suggestion
require_once __DIR__ . "/../../../config/centreon.config.php";
```

File: www/install/step_upgrade/step3.php [hunk 2]
```diff
@@ -43,55 +43,59 @@
...
+include_once _CENTREON_PATH_ . "/www/class/centreonDB.class.php";
```
Comment (kduret) [inline www/install/step_upgrade/step3.php:47]:
```suggestion
require_once __DIR__ . "/../../class/centreonDB.class.php";
```

File: www/install/step_upgrade/step3.php [hunk 3]
```diff
@@ -43,55 +43,59 @@
...
+                (empty($next) || (version_compare($next, $matches[1]) < 0))) {
+                $next = $matches[1];
+            }
+        }
     }
-    $contents = "<textarea cols='100' rows='30' readonly>" . $releasenotesContent . "</textarea>";
+    closedir($handle);
 }
 
+$majors = explode('.', $next);
```
Comment (kduret) [inline www/install/step_upgrade/step3.php:77]: prefer usage of preg_match : `preg_match('/^(\d+\.\d+)/', $next, $matches);`
Then, $matches[1] will be the major version (eg: 19.04)

File: www/install/step_upgrade/step3.php [hunk 4]
```diff
@@ -43,55 +43,59 @@
...
+/*
+** To find the final version that we should update to, we will look in
+** the www/install/php directory where all PHP update scripts are
+** stored. We will extract the target version from the filename and find
+** the farthest version to the current version.
+*/
+$next = '';
+if ($handle = opendir('../php')) {
+    while (false !== ($file = readdir($handle))) {
+        if (preg_match('/Update-([a-zA-Z0-9\-\.]+)\.php/', $file, $matches)) {
```
Comment (kduret) [inline www/install/step_upgrade/step3.php:67]:
```suggestion
        if (preg_match('/Update-(\d+\.\d+\.\d+)\.php/', $file, $matches)) {
```

File: www/install/step_upgrade/step3.php [hunk 5]
```diff
@@ -43,55 +43,59 @@
...
+
 $template->assign('step', STEP_NUMBER);
 $template->assign('title', $title);
 $template->assign('content', $contents);
 $template->assign('blockPreview', 1);
 $template->display('content.tpl');
 ?>
 <script type='text/javascript'>
-    var step3_s = 10;
+    var step3_s = 3;
```
Comment (kduret) [inline www/install/step_upgrade/step3.php:95]: do we still need this timer ?

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -3532,6 +3532,18 @@ msgstr ""
 msgid "Release notes"
 msgstr "Notes de version"
 
+#: centreon-web/www/install/step_upgrade/step3.php:46
+msgid "Everything is ready !"
+msgstr "Tout est prêts !"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:3537]:
```suggestion
msgstr "Tout est prêt !"
```

File: www/install/step_upgrade/step3.php (hunk 5 above)
Comment (adr-mo) [reply www/install/step_upgrade/step3.php:95]: In my opinion no, but I've followed the written story. @cgagnaire maybe we could get rid of this ?

File: www/install/step_upgrade/step3.php (hunk 3 above)
Comment (adr-mo) [reply www/install/step_upgrade/step3.php]: done

File: www/install/step_upgrade/step3.php (hunk 5 above)
Comment (cgagnaire) [reply www/install/step_upgrade/step3.php:95]: Ask @Moujimouja :)

File: www/install/step_upgrade/step3.php (hunk 4 above)
Comment (adr-mo) [reply www/install/step_upgrade/step3.php:67]: This will not work with versions like "20.04.0-rc2" no ?

Comment (kduret) [reply www/install/step_upgrade/step3.php:67]: indeed

File: www/install/step_upgrade/step3.php (hunk 5 above)
Comment (adr-mo) [reply www/install/step_upgrade/step3.php:95]: I guess we will keep it for now, waiting for the build to be :green_heart: to merge

**Pull request merged**